components/support/rc_crypto/src/contentsignature.rs

```diff
@@ -0,0 +1,305 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::str;
+
+use crate::error::*;
+use crate::signature;
+
+fn decode_root_hash(input: &str) -> Result<Vec<u8>> {
+    let bytes_hex = input.split(':');
```

Comment:
An example of the sort of string we're trying to pass here would be useful to help understanding, I am interpreting it as something like "ab:12:ef:...".